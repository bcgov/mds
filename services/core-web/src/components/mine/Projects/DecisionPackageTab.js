// TODO: Line 106 - Replace <Alert /> with a dynamic one that updates based on status code

import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators, compose } from "redux";
import { getFormValues } from "redux-form";
import { withRouter } from "react-router-dom";
import { Row, Col, Typography, Button } from "antd";
import { LockOutlined, FolderViewOutlined } from "@ant-design/icons";
import PropTypes from "prop-types";
import { getProject } from "@common/selectors/projectSelectors";
import { openModal, closeModal } from "@common/actions/modalActions";
import {
  fetchProjectById,
  createProjectDecisionPackage,
  updateProjectDecisionPackage,
  removeDocumentFromProjectDecisionPackage,
} from "@common/actionCreators/projectActionCreator";
import { EDIT_OUTLINE_VIOLET } from "@/constants/assets";
import * as routes from "@/constants/routes";
import customPropTypes from "@/customPropTypes";
import ScrollSideMenu from "@/components/common/ScrollSideMenu";
import DocumentTable from "@/components/common/DocumentTable";
import UpdateDecisionPackageStatusForm from "@/components/Forms/majorMineApplication/UpdateDecisionPackageStatusForm";
import { modalConfig } from "@/components/modalContent/config";
import { getProjectDecisionPackageStatusCodesHash } from "@common/selectors/staticContentSelectors";
import * as FORM from "@/constants/forms";

const propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      projectGuid: PropTypes.string,
    }),
  }).isRequired,
  project: customPropTypes.project.isRequired,
  fetchProjectById: PropTypes.func.isRequired,
  openModal: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  updateProjectDecisionPackage: PropTypes.func.isRequired,
  createProjectDecisionPackage: PropTypes.func.isRequired,
  removeDocumentFromProjectDecisionPackage: PropTypes.func.isRequired,
  projectDecisionPackageStatusCodesHash: PropTypes.objectOf(PropTypes.string).isRequired,
  formValues: PropTypes.objectOf(PropTypes.any).isRequired,
};

export class DecisionPackageTab extends Component {
  state = {
    fixedTop: false,
  };

  componentDidMount() {
    this.handleFetchData();
    window.addEventListener("scroll", this.handleScroll);
    this.handleScroll();
  }

  componentWillUnmount() {
    window.removeEventListener("scroll", this.handleScroll);
  }

  handleScroll = () => {
    if (window.pageYOffset > 170 && !this.state.fixedTop) {
      this.setState({ fixedTop: true });
    } else if (window.pageYOffset <= 170 && this.state.fixedTop) {
      this.setState({ fixedTop: false });
    }
  };

  handleFetchData = () => {
    const { projectGuid } = this.props.match?.params;
    return this.props.fetchProjectById(projectGuid);
  };

  handleUpdateProjectDecisionPackage = (event, values) => {
    event.preventDefault();
    const { projectGuid } = this.props.match?.params;
    const projectDecisionPackage = this.props.project.project_decision_package;
    const projectDecisionPackageGuid = projectDecisionPackage?.project_decision_package_guid;

    if (!projectDecisionPackageGuid) {
      return this.props
        .createProjectDecisionPackage(
          {
            projectGuid: this.props.project?.project_guid,
          },
          values
        )
        .then(() => this.handleFetchData());
    }
    return this.props
      .updateProjectDecisionPackage(
        {
          projectGuid,
          projectDecisionPackageGuid,
        },
        values
      )
      .then(() => this.handleFetchData());
  };

  handleDeleteDocument = (event, documentKey) => {
    event.preventDefault();
    const { project_guid: projectGuid, project_decision_package } = this.props.project;

    return this.props
      .removeDocumentFromProjectDecisionPackage(
        projectGuid,
        project_decision_package?.project_decision_package_guid,
        documentKey
      )
      .then(() => this.handleFetchData());
  };

  renderDocumentSection = (sectionTitle, sectionHref, sectionText, sectionDocuments) => {
    const titleElement = (
      <Typography.Text strong style={{ fontSize: "1.5rem" }}>
        {sectionTitle}
      </Typography.Text>
    );

    return (
      <div id={sectionHref}>
        <p>{titleElement}</p>
        <br />
        <p>{sectionText}</p>
        <DocumentTable
          documents={sectionDocuments?.reduce(
            (docs, doc) => [
              {
                key: doc.mine_document_guid,
                mine_document_guid: doc.mine_document_guid,
                document_manager_guid: doc.document_manager_guid,
                name: doc.document_name,
                category: null,
                uploaded: doc.upload_date,
              },
              ...docs,
            ],
            []
          )}
          excludedColumnKeys={["category"]}
          additionalColumnProps={[{ key: "name", colProps: { width: "80%" } }]}
          removeDocument={this.handleDeleteDocument}
        />
      </div>
    );
  };

  handleOpenModal = (modalType) => {
    let title;
    let contentTitle;
    let instructions;
    let content = modalConfig.UPLOAD_PROJECT_DECISION_PACKAGE_DOCUMENT_MODAL;
    let submitHandler = this.handleUploadDocument;
    let optionalProps = {};
    if (modalType === "upload-document") {
      title = "Upload Documents";
      instructions =
        "Please upload all relevant decision documentation below. You can add this set of files directly to your decision package by selecting the option below.";
      contentTitle = "Upload Documents";
    } else if (modalType === "internal") {
      title = "Upload Internal Documents";
      instructions =
        "Upload internal documents that are created durring the review process. These files are for internal staff only and will not be shown to proponents.";
      contentTitle = "Upload Internal Ministry Document";
    } else if (modalType === "edit-decision-package") {
      content = modalConfig.UPDATE_PROJECT_DECISION_PACKAGE_DOCUMENT_MODAL;
      submitHandler = this.handleUpdateProjectDecisionPackage;
      optionalProps = {
        documents: this.props.project.project_decision_package?.documents,
        status_code: this.props.project.project_decision_package?.status_code,
      };
    }

    return this.props.openModal({
      props: {
        title,
        contentTitle,
        instructions,
        projectGuid: this.props.project?.project_guid,
        modalType,
        closeModal: this.props.closeModal,
        handleSubmit: submitHandler,
        afterClose: () => {},
        optionalProps,
      },
      content,
    });
  };

  handleUploadDocument = (event, values, flags) => {
    const payload = {
      documents: values.map((doc) => {
        // Determine document type
        const { isDecisionPackageEligible, addFilesToDecisionPackage } = flags;
        let project_decision_package_document_type_code;
        if (isDecisionPackageEligible && addFilesToDecisionPackage) {
          project_decision_package_document_type_code = "DCP";
        } else if (isDecisionPackageEligible && !addFilesToDecisionPackage) {
          project_decision_package_document_type_code = "ADG";
        } else {
          project_decision_package_document_type_code = "INM";
        }
        return {
          ...doc,
          project_decision_package_document_type_code,
        };
      }),
      status_code: this.props.formValues?.status_code,
    };
    this.handleUpdateProjectDecisionPackage(event, payload);
    return this.props.closeModal();
  };

  render() {
    const projectDecisionPackage = this.props.project.project_decision_package;
    const allDocuments = projectDecisionPackage?.documents;
    const hasStartedPackage =
      Boolean(projectDecisionPackage?.project_decision_package_guid) &&
      projectDecisionPackage?.status_code !== "NTS";

    return (
      <>
        <div className={this.state.fixedTop ? "side-menu--fixed" : "side-menu"}>
          <ScrollSideMenu
            menuOptions={[
              { href: "decision-package-documents", title: "Decision Package" },
              { href: "additional-goverment-documents", title: "Government Documents" },
              { href: "internal-ministry-documents", title: "Internal Documents" },
            ]}
            featureUrlRoute={routes.PROJECT_DECISION_PACKAGE.hashRoute}
            featureUrlRouteArguments={[this.props.match?.params?.projectGuid]}
          />
        </div>
        <div
          className={
            this.state.fixedTop ? "side-menu--content with-fixed-top top-125" : "side-menu--content"
          }
        >
          <Row>
            <UpdateDecisionPackageStatusForm
              initialValues={{
                status_code: projectDecisionPackage?.status_code || "NTS",
              }}
              displayValues={{
                status_code: projectDecisionPackage?.status_code || "NTS",
                projectDecisionPackageStatusCodesHash: this.props
                  ?.projectDecisionPackageStatusCodesHash,
                updateUser: projectDecisionPackage?.update_user,
                updateDate: projectDecisionPackage?.update_timestamp,
                documents: projectDecisionPackage?.documents,
              }}
              handleSubmit={this.handleUpdateProjectDecisionPackage}
            />
            <Col span={24}>
              <Typography.Title level={3}>
                <br />
                <FolderViewOutlined className="violet" />
                &nbsp;Decision Package (Proponent Visible)
              </Typography.Title>
              <hr />
            </Col>
          </Row>
          {this.renderDocumentSection(
            <Row>
              <Col xs={24} md={12}>
                Decision Package Documents
              </Col>
              <Col xs={24} md={12}>
                <Button
                  type="primary"
                  style={{ float: "right" }}
                  disabled={!hasStartedPackage}
                  onClick={() => this.handleOpenModal("upload-document")}
                >
                  + Add Documents
                </Button>
                <Button
                  type="secondary"
                  style={{ float: "right" }}
                  disabled={!hasStartedPackage || allDocuments?.length === 0}
                  onClick={() => this.handleOpenModal("edit-decision-package")}
                >
                  <img name="edit" src={EDIT_OUTLINE_VIOLET} alt="Edit" />
                  &nbsp; Edit Package
                </Button>
              </Col>
            </Row>,
            "decision-package-documents",
            <Typography.Text>
              <b>These files are visible to the proponent.</b> Upload Ministry Decision
              documentation.
            </Typography.Text>,
            allDocuments?.filter(
              (doc) => doc.project_decision_package_document_type_code === "DCP"
            ) || []
          )}
          <br />
          {this.renderDocumentSection(
            "Additional Government Documents",
            "additional-goverment-documents",
            <Typography.Text>
              <b>These files are visible to the proponent.</b> Upload Supplemental Ministry
              documentation.
            </Typography.Text>,
            allDocuments?.filter(
              (doc) => doc.project_decision_package_document_type_code === "ADG"
            ) || []
          )}
          <br />
          <Row>
            <Col span={24}>
              <Typography.Title level={3}>
                <LockOutlined className="violet" />
                &nbsp;Confidential Internal Documents (Ministry Visible Only)
              </Typography.Title>
              <hr />
            </Col>
          </Row>
          {this.renderDocumentSection(
            <Row>
              <Col xs={24} md={12}>
                Internal Ministry Documentation
              </Col>
              <Col xs={24} md={12}>
                <Button
                  type="primary"
                  style={{ float: "right" }}
                  disabled={!hasStartedPackage}
                  onClick={() => this.handleOpenModal("internal")}
                >
                  + Add Documents
                </Button>
              </Col>
            </Row>,
            "internal-ministry-documents",
            <Typography.Text>
              <b>These files are for internal staff only and will not be shown to proponents.</b>{" "}
              Upload internal documents that are created durring the review process.
            </Typography.Text>,
            allDocuments?.filter(
              (doc) => doc.project_decision_package_document_type_code === "INM"
            ) || []
          )}
        </div>
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  project: getProject(state),
  projectDecisionPackageStatusCodesHash: getProjectDecisionPackageStatusCodesHash(state),
  formValues: getFormValues(FORM.UPDATE_PROJECT_DECISION_PACKAGE)(state) || {},
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      updateProjectDecisionPackage,
      fetchProjectById,
      createProjectDecisionPackage,
      removeDocumentFromProjectDecisionPackage,
      openModal,
      closeModal,
    },
    dispatch
  );

DecisionPackageTab.propTypes = propTypes;

export default compose(
  withRouter,
  connect(mapStateToProps, mapDispatchToProps)
)(DecisionPackageTab);
