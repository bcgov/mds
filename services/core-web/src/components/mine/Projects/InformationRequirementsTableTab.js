import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { Tabs, Tag } from "antd";
import "@ant-design/compatible/assets/index.css";
import PropTypes from "prop-types";
import { formatDate } from "@common/utils/helpers";
import {
  fetchRequirements,
  fetchProjectById,
  updateInformationRequirementsTable,
} from "@common/actionCreators/projectActionCreator";
import { getInformationRequirementsTableStatusCodesHash } from "@common/selectors/staticContentSelectors";
import {
  getProject,
  getInformationRequirementsTable,
  getRequirements,
} from "@common/selectors/projectSelectors";
import { ArrowLeftOutlined, EnvironmentOutlined } from "@ant-design/icons";
import ReviewInformationRequirementsTable from "@/components/mine/Projects/ReviewInformationRequirementsTable";
import UpdateInformationRequirementsTableForm from "@/components/Forms/informationRequirementsTable/UpdateInformationRequirementsTableForm";
import CustomPropTypes from "@/customPropTypes";
import * as routes from "@/constants/routes";

const propTypes = {
  project: CustomPropTypes.project.isRequired,
  requirements: PropTypes.arrayOf(CustomPropTypes.requirements).isRequired,
  informationRequirementsTable: CustomPropTypes.informationRequirementsTable.isRequired,
  informationRequirementsTableStatusCodesHash: PropTypes.objectOf(PropTypes.string).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      projectGuid: PropTypes.string,
      irtGuid: PropTypes.string,
    }),
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
  fetchProjectById: PropTypes.func.isRequired,
  fetchRequirements: PropTypes.func.isRequired,
  updateInformationRequirementsTable: PropTypes.func.isRequired,
};

const sideMenuOptions = [
  {
    href: "intro-project-overview",
    title: "Introduction and Project Overview",
  },
  {
    href: "baseline-information",
    title: "Baseline Information",
  },
  {
    href: "mine-plan",
    title: "Mine Plan",
  },
  {
    href: "reclamation-closure-plan",
    title: "Reclamation Closure Plan",
  },
  {
    href: "modelling-mitigation-discharges",
    title: "Modelling Mitigation Discharges",
  },
  {
    href: "environmental-assessment-predictions",
    title: "Environmental Assessment Predictions",
  },
  {
    href: "environmental-monitoring",
    title: "Environmental Monitory",
  },
  {
    href: "health-safety",
    title: "Health Safety",
  },
  {
    href: "management-plan",
    title: "Management Plan",
  },
];

export class InformationRequirementsTableTab extends Component {
  mergedRequirements = [];

  constructor(props) {
    super(props);
    this.state = {
      fixedTop: false,
      activeTab: "intro-project-overview",
    };
  }

  componentDidMount() {
    this.handleFetchData();
    window.addEventListener("scroll", this.handleScroll);
    this.handleScroll();
  }

  componentWillUnmount() {
    window.removeEventListener("scroll", this.handleScroll);
  }

  /* eslint-disable no-param-reassign */
  deepMergeById = (r1, r2) =>
    r1.map(({ requirement_guid, sub_requirements, ...rest }) => ({
      requirement_guid,
      ...rest,
      ...r2.find((i) => i.requirement_guid === requirement_guid),
      sub_requirements: this.deepMergeById(sub_requirements, r2),
    }));

  handleScroll = () => {
    if (window.pageYOffset > 170 && !this.state.fixedTop) {
      this.setState({ fixedTop: true });
    } else if (window.pageYOffset <= 170 && this.state.fixedTop) {
      this.setState({ fixedTop: false });
    }
  };

  handleTabChange = (activeTab) => {
    const { projectGuid, irtGuid } = this.props.match.params;
    const url = routes.INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(projectGuid, irtGuid, activeTab);
    this.props.history.push(url);
    this.setState({ activeTab });
  };

  handleFetchData = () => {
    const { projectGuid } = this.props.match.params;
    if (projectGuid) {
      return this.props
        .fetchProjectById(projectGuid)
        .then(() => this.props.fetchRequirements())
        .then(() => this.setState({ isLoaded: true }))
        .catch(() => this.setState({ isLoaded: false }));
    }
    return null;
  };

  handleUpdateIRT = (event, values) => {
    event.preventDefault();
    const { projectGuid, irtGuid } = this.props.match.params;
    return this.props
      .updateInformationRequirementsTable(
        {
          projectGuid,
          informationRequirementsTableGuid: irtGuid,
        },
        values
      )
      .then(() => this.handleFetchData());
  };

  render() {
    const {
      project_title,
      mine_name,
      mine_guid,
      project_guid,
      information_requirements_table,
    } = this.props.project;
    const updateUser = information_requirements_table?.update_user;
    const updateDate = formatDate(information_requirements_table?.update_timestamp);
    const statusCode = information_requirements_table?.status_code;

    this.mergedRequirements = this.deepMergeById(
      this.props.requirements,
      this.props.informationRequirementsTable.requirements
    );

    return (
      <div className="page">
        <div
          className={
            this.state.fixedTop
              ? "padding-lg view--header fixed-scroll"
              : " padding-lg view--header"
          }
        >
          <h1>
            {project_title}
            <span className="padding-sm--left">
              <Tag title={`Mine: ${mine_name}`}>
                <Link
                  style={{ textDecoration: "none" }}
                  to={routes.MINE_GENERAL.dynamicRoute(mine_guid)}
                  disabled={!mine_guid}
                >
                  <EnvironmentOutlined className="padding-sm--right" />
                  {mine_name}
                </Link>
              </Tag>
            </span>
          </h1>
          <Link to={routes.PROJECTS.dynamicRoute(project_guid)}>
            <ArrowLeftOutlined className="padding-sm--right" />
            Back to: {mine_name} Project overview
          </Link>
          <Tabs
            size="large"
            animated={{ inkBar: true, tabPane: false }}
            className="now-tabs"
            style={{ margin: "0" }}
            centered
          >
            <Tabs.TabPane tab="IRT" key="irt">
              <></>
            </Tabs.TabPane>
          </Tabs>
        </div>
        <div>
          <Tabs
            size="large"
            activeKey={this.state.activeTab}
            className="now-tabs vertical-tabs"
            tabPosition="left"
            animated={{ inkBar: true, tabPane: false }}
            onChange={(tab) => this.handleTabChange(tab)}
          >
            {sideMenuOptions.map((tab, idx) => (
              <Tabs.TabPane tab={tab.title} key={tab.href} className="vertical-tabs--tabpane">
                <div>
                  <UpdateInformationRequirementsTableForm
                    initialValues={{
                      status_code: statusCode,
                    }}
                    displayValues={{
                      statusCode,
                      updateUser,
                      updateDate,
                      informationRequirementsTableStatusCodesHash: this.props
                        .informationRequirementsTableStatusCodesHash,
                    }}
                    handleSubmit={this.handleUpdateIRT}
                  />
                  <ReviewInformationRequirementsTable
                    requirements={this.mergedRequirements[idx]}
                    isLoaded={this.state.isLoaded}
                  />
                </div>
              </Tabs.TabPane>
            ))}
          </Tabs>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  project: getProject(state),
  informationRequirementsTable: getInformationRequirementsTable(state),
  requirements: getRequirements(state),
  informationRequirementsTableStatusCodesHash: getInformationRequirementsTableStatusCodesHash(
    state
  ),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchProjectById,
      fetchRequirements,
      updateInformationRequirementsTable,
    },
    dispatch
  );

InformationRequirementsTableTab.propTypes = propTypes;

export default connect(mapStateToProps, mapDispatchToProps)(InformationRequirementsTableTab);
