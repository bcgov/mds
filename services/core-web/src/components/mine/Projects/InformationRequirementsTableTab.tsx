import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useParams } from "react-router-dom";
import { Tabs } from "antd";
import { formatDate } from "@mds/common/redux/utils/helpers";
import {
  fetchRequirements,
  fetchProjectById,
  updateInformationRequirementsTable,
} from "@mds/common/redux/actionCreators/projectActionCreator";
import { getInformationRequirementsTableStatusCodesHash } from "@mds/common/redux/selectors/staticContentSelectors";
import {
  getProject,
  getInformationRequirementsTable,
  getRequirements,
} from "@mds/common/redux/selectors/projectSelectors";
import ReviewInformationRequirementsTable from "@/components/mine/Projects/ReviewInformationRequirementsTable";
import UpdateInformationRequirementsTableForm from "@/components/Forms/informationRequirementsTable/UpdateInformationRequirementsTableForm";

const sideMenuOptions = [
  {
    href: "intro-project-overview",
    title: "Introduction and Project Overview",
  },
  {
    href: "baseline-information",
    title: "Baseline Information",
  },
  {
    href: "mine-plan",
    title: "Mine Plan",
  },
  {
    href: "reclamation-closure-plan",
    title: "Reclamation Closure Plan",
  },
  {
    href: "modelling-mitigation-discharges",
    title: "Modelling Mitigation Discharges",
  },
  {
    href: "environmental-assessment-predictions",
    title: "Environmental Assessment Predictions",
  },
  {
    href: "environmental-monitoring",
    title: "Environmental Monitory",
  },
  {
    href: "health-safety",
    title: "Health Safety",
  },
  {
    href: "management-plan",
    title: "Management Plan",
  },
];

const InformationRequirementsTableTab = () => {
  const { projectGuid } = useParams<{ projectGuid: string }>();
  const dispatch = useDispatch();

  const project = useSelector(getProject);
  const requirements = useSelector(getRequirements);
  const informationRequirementsTable = useSelector(getInformationRequirementsTable);
  const informationRequirementsTableStatusCodesHash = useSelector(
    getInformationRequirementsTableStatusCodesHash
  );

  const [isLoaded, setIsLoaded] = useState(false);

  const handleFetchData = () => {
    dispatch(fetchProjectById(projectGuid))
      .then(() => dispatch(fetchRequirements()))
      .catch(setIsLoaded(false));
  };

  useEffect(() => {
    handleFetchData();
  }, []);

  useEffect(() => {
    if (
      projectGuid &&
      project.project_guid === projectGuid &&
      informationRequirementsTable?.project_guid === projectGuid
    ) {
      setIsLoaded(true);
    }
  }, [project, informationRequirementsTable]);

  const deepMergeById = (r1, r2) =>
    r1.map(({ requirement_guid, sub_requirements, ...rest }) => ({
      requirement_guid,
      ...rest,
      ...r2.find((i) => i.requirement_guid === requirement_guid),
      sub_requirements: deepMergeById(sub_requirements, r2),
    }));

  const handleUpdateIRT = (event, values) => {
    event.preventDefault();
    dispatch(
      updateInformationRequirementsTable(
        { projectGuid, informationRequirementsTableGuid: informationRequirementsTable.irt_guid },
        values
      )
    ).then(() => handleFetchData());
  };

  const { information_requirements_table } = project;
  const updateUser = information_requirements_table?.update_user;
  const updateDate = formatDate(information_requirements_table?.update_timestamp);
  const statusCode = information_requirements_table?.status_code;

  const mergedRequirements = deepMergeById(
    requirements,
    informationRequirementsTable?.requirements?.filter(({ deleted_ind }) => deleted_ind === false)
  );

  return (
    <div>
      <Tabs
        size="large"
        defaultActiveKey={sideMenuOptions[0].href}
        className="now-tabs vertical-tabs"
        tabPosition="left"
        animated={{ inkBar: true, tabPane: false }}
        items={sideMenuOptions.map((tab, idx) => ({
          key: tab.href,
          label: tab.title,
          children: (
            <div>
              <UpdateInformationRequirementsTableForm
                initialValues={{
                  status_code: statusCode,
                }}
                displayValues={{
                  statusCode,
                  updateUser,
                  updateDate,
                  informationRequirementsTableStatusCodesHash: informationRequirementsTableStatusCodesHash,
                }}
                handleSubmit={handleUpdateIRT}
              />
              <ReviewInformationRequirementsTable
                requirements={mergedRequirements[idx]}
                isLoaded={isLoaded}
              />
            </div>
          ),
        }))}
      />
    </div>
  );
};
export default InformationRequirementsTableTab;
