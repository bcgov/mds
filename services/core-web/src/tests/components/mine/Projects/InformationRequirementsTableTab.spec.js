import React from "react";
import { shallow } from "enzyme";
import { InformationRequirementsTableTab } from "@/components/mine/Projects/InformationRequirementsTableTab";
import * as MOCK from "@/tests/mocks/dataMocks";

const props = {};
const dispatchProps = {};

const setupProps = () => {
  props.project = MOCK.PROJECT;
  props.informationRequirementsTable = MOCK.INFORMATION_REQUIREMENTS_TABLE;
  props.requirements = MOCK.REQUIREMENTS.records;
  props.match = { params: { projectGuid: "123", irtGuid: "456" } };
  props.history = { push: jest.fn() };
};

const setupDispatchProps = () => {
  dispatchProps.fetchRequirements = jest.fn(() => Promise.resolve());
  dispatchProps.fetchProjectById = jest.fn(() => Promise.resolve());
};

beforeEach(() => {
  setupProps();
  setupDispatchProps();
});

describe("InformationRequirementsTableTab", () => {
  it("renders properly", () => {
    const component = shallow(<InformationRequirementsTableTab {...props} {...dispatchProps} />);
    expect(component).toMatchSnapshot();
  });
});
