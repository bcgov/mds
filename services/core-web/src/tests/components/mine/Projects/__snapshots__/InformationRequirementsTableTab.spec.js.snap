// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationRequirementsTableTab renders properly 1`] = `
<div
  className="page"
>
  <div
    className=" padding-lg view--header"
  >
    <h1>
      Test Project Title
      <span
        className="padding-sm--left"
      >
        <Tag
          title="Mine: Sample Mine"
        >
          <Link
            disabled={false}
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to="/mine-dashboard/40fb0ca4-4dfb-4660-a184-6d031a21f3e9/mine-information/general"
          >
            <ForwardRef(EnvironmentOutlined)
              className="padding-sm--right"
            />
            Sample Mine
          </Link>
        </Tag>
      </span>
    </h1>
    <Link
      to="/pre-applications/35633148-57f8-4967-be35-7f89abfbd02e/overview"
    >
      <ForwardRef(ArrowLeftOutlined)
        className="padding-sm--right"
      />
      Back to: 
      Sample Mine
       Project overview
    </Link>
    <Tabs
      animated={
        Object {
          "inkBar": true,
          "tabPane": false,
        }
      }
      centered={true}
      className="now-tabs"
      size="large"
      style={
        Object {
          "margin": "0",
        }
      }
    >
      <TabPane
        key="irt"
        tab="IRT"
      />
    </Tabs>
  </div>
  <div>
    <Tabs
      activeKey="intro-project-overview"
      animated={
        Object {
          "inkBar": true,
          "tabPane": false,
        }
      }
      className="now-tabs vertical-tabs"
      onChange={[Function]}
      size="large"
      tabPosition="left"
    >
      <TabPane
        className="vertical-tabs--tabpane"
        key="intro-project-overview"
        tab="Introduction and Project Overview"
      >
        <div>
          <Connect(ReduxForm)
            displayValues={
              Object {
                "informationRequirementsTableStatusCodesHash": undefined,
                "statusCode": undefined,
                "updateDate": undefined,
                "updateUser": undefined,
              }
            }
            handleSubmit={[Function]}
            initialValues={
              Object {
                "status_code": undefined,
              }
            }
          />
          <ReviewInformationRequirementsTable
            requirements={
              Object {
                "deleted_ind": false,
                "description": "Introduction and Project Overview",
                "display_order": 1,
                "parent_requirement_id": null,
                "requirement_guid": "d9996a84-30b0-4de6-b171-c959bdae0e3f",
                "requirement_id": 1,
                "step": "1.",
                "sub_requirements": Array [
                  Object {
                    "deleted_ind": false,
                    "description": "Project Overview",
                    "display_order": 3,
                    "parent_requirement_id": 1,
                    "requirement_guid": "56e58309-9f63-401c-b26d-bfdc867ebe5c",
                    "requirement_id": 12,
                    "step": "1.3.",
                    "sub_requirements": Array [
                      Object {
                        "deleted_ind": false,
                        "description": "Project History",
                        "display_order": 1,
                        "parent_requirement_id": 12,
                        "requirement_guid": "9e43699d-da0c-49e4-849e-18fc74d33a4f",
                        "requirement_id": 78,
                        "step": "1.3.1.",
                        "sub_requirements": Array [],
                      },
                    ],
                  },
                  Object {
                    "deleted_ind": false,
                    "description": "Regulatory Framework",
                    "display_order": 4,
                    "parent_requirement_id": 1,
                    "requirement_guid": "808a2f93-9774-49ae-bc85-82be8ce3ea2f",
                    "requirement_id": 13,
                    "step": "1.4.",
                    "sub_requirements": Array [],
                  },
                ],
              }
            }
          />
        </div>
      </TabPane>
      <TabPane
        className="vertical-tabs--tabpane"
        key="baseline-information"
        tab="Baseline Information"
      >
        <div>
          <Connect(ReduxForm)
            displayValues={
              Object {
                "informationRequirementsTableStatusCodesHash": undefined,
                "statusCode": undefined,
                "updateDate": undefined,
                "updateUser": undefined,
              }
            }
            handleSubmit={[Function]}
            initialValues={
              Object {
                "status_code": undefined,
              }
            }
          />
          <ReviewInformationRequirementsTable />
        </div>
      </TabPane>
      <TabPane
        className="vertical-tabs--tabpane"
        key="mine-plan"
        tab="Mine Plan"
      >
        <div>
          <Connect(ReduxForm)
            displayValues={
              Object {
                "informationRequirementsTableStatusCodesHash": undefined,
                "statusCode": undefined,
                "updateDate": undefined,
                "updateUser": undefined,
              }
            }
            handleSubmit={[Function]}
            initialValues={
              Object {
                "status_code": undefined,
              }
            }
          />
          <ReviewInformationRequirementsTable />
        </div>
      </TabPane>
      <TabPane
        className="vertical-tabs--tabpane"
        key="reclamation-closure-plan"
        tab="Reclamation Closure Plan"
      >
        <div>
          <Connect(ReduxForm)
            displayValues={
              Object {
                "informationRequirementsTableStatusCodesHash": undefined,
                "statusCode": undefined,
                "updateDate": undefined,
                "updateUser": undefined,
              }
            }
            handleSubmit={[Function]}
            initialValues={
              Object {
                "status_code": undefined,
              }
            }
          />
          <ReviewInformationRequirementsTable />
        </div>
      </TabPane>
      <TabPane
        className="vertical-tabs--tabpane"
        key="modelling-mitigation-discharges"
        tab="Modelling Mitigation Discharges"
      >
        <div>
          <Connect(ReduxForm)
            displayValues={
              Object {
                "informationRequirementsTableStatusCodesHash": undefined,
                "statusCode": undefined,
                "updateDate": undefined,
                "updateUser": undefined,
              }
            }
            handleSubmit={[Function]}
            initialValues={
              Object {
                "status_code": undefined,
              }
            }
          />
          <ReviewInformationRequirementsTable />
        </div>
      </TabPane>
      <TabPane
        className="vertical-tabs--tabpane"
        key="environmental-assessment-predictions"
        tab="Environmental Assessment Predictions"
      >
        <div>
          <Connect(ReduxForm)
            displayValues={
              Object {
                "informationRequirementsTableStatusCodesHash": undefined,
                "statusCode": undefined,
                "updateDate": undefined,
                "updateUser": undefined,
              }
            }
            handleSubmit={[Function]}
            initialValues={
              Object {
                "status_code": undefined,
              }
            }
          />
          <ReviewInformationRequirementsTable />
        </div>
      </TabPane>
      <TabPane
        className="vertical-tabs--tabpane"
        key="environmental-monitoring"
        tab="Environmental Monitory"
      >
        <div>
          <Connect(ReduxForm)
            displayValues={
              Object {
                "informationRequirementsTableStatusCodesHash": undefined,
                "statusCode": undefined,
                "updateDate": undefined,
                "updateUser": undefined,
              }
            }
            handleSubmit={[Function]}
            initialValues={
              Object {
                "status_code": undefined,
              }
            }
          />
          <ReviewInformationRequirementsTable />
        </div>
      </TabPane>
      <TabPane
        className="vertical-tabs--tabpane"
        key="health-safety"
        tab="Health Safety"
      >
        <div>
          <Connect(ReduxForm)
            displayValues={
              Object {
                "informationRequirementsTableStatusCodesHash": undefined,
                "statusCode": undefined,
                "updateDate": undefined,
                "updateUser": undefined,
              }
            }
            handleSubmit={[Function]}
            initialValues={
              Object {
                "status_code": undefined,
              }
            }
          />
          <ReviewInformationRequirementsTable />
        </div>
      </TabPane>
      <TabPane
        className="vertical-tabs--tabpane"
        key="management-plan"
        tab="Management Plan"
      >
        <div>
          <Connect(ReduxForm)
            displayValues={
              Object {
                "informationRequirementsTableStatusCodesHash": undefined,
                "statusCode": undefined,
                "updateDate": undefined,
                "updateUser": undefined,
              }
            }
            handleSubmit={[Function]}
            initialValues={
              Object {
                "status_code": undefined,
              }
            }
          />
          <ReviewInformationRequirementsTable />
        </div>
      </TabPane>
    </Tabs>
  </div>
</div>
`;
